railties/lib/rails/generators/rails/app/templates/Procfile.tt

```diff
@@ -0,0 +1,4 @@
+web: bundle exec bin/rails server
```

Comment:
Do we really need the `bundle exec`? I searched the code base and I found a lot of instances of `bin/rails server` but none of them had `bundle exec` prefixed. 
Actually, I can't find a single instance of `bundle exec bin/rails`.

Reply:
@sharang-d: that's right, we have changed every place in the code to run commands with `bin/rails` instead of `bundle exec rails` or `bundle exec bin/rails`, so I think we should do it here also

Reply:
I think the reasoning here would be, on deployed platforms this is preferred way to do this?
Most Procfiles I have used, make use of `bundle exec` to make them work properly when deployed on systems.

Reply:
I was under the impression that we need to do either 
```
bundle exec rails ...
``` 
(Old style)
or 
```
bin/rails ...
```
(New style)
.

This one has both :thinking:.

Reply:
The bundler integration with RubyGems broke command execution for Ruby 2.6.1 and it won't be fixed until Ruby 2.6.2 https://bugs.ruby-lang.org/issues/15622. The fix is to preface commands with `bundle exec` https://devcenter.heroku.com/articles/bundler-version#known-upgrade-issues.

While this is a very unfortunate bug, if we want this to work out of the box with Ruby 2.6.1 then `bundle exec` is sadly needed. Once 2.6.2 comes out we can drop it assuming there's no future regressions.

Reply:
Got it. Thanks for the info! 🙂